python/lsst/daf/butler/_butler.py

```diff
@@ -987,6 +987,31 @@ def find_dataset(
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def retrieve_artifacts_zip(
+        self,
+        refs: Iterable[DatasetRef],
+        destination: ResourcePathExpression,
+    ) -> ResourcePath:
+        """Retrieve artifacts from a Butler and place in ZIP file.
+
+        Parameters
+        ----------
+        refs : `collections.abc.Iterable` [ `DatasetRef` ]
```

Comment:
```suggestion
        refs : `~collections.abc.Iterable` [ `DatasetRef` ]
```

What happens if iterable is empty, will it create an empty ZIP file or raise an exception?

Reply:
Currently you get a Zip file with an index in it. Maybe it should raise? Not sure.

Reply:
I think raising is probably better in that case, who would want an empty ZIp?